src/api/ZoweExplorerZosmfApi.ts

```diff
@@ -36,23 +38,134 @@ class ZosmfApiCommon implements ZoweExplorerApi.ICommon {
         return ZosmfUssApi.getProfileTypeName();
     }
 
-    public getSession(profile?: IProfileLoaded): Session {
+    public async getSession(profile?: IProfileLoaded, prompt?: boolean): Promise<Session> {
         if (!this.session) {
-            this.session = zowe.ZosmfSession.createBasicZosmfSession((profile||this.profile).profile);
+            this.session = await this.getValidSession((profile||this.profile), (profile||this.profile).name, (prompt||false));
         }
         return this.session;
     }
 
-    public async getStatus(validateProfile?: IProfileLoaded, profileType?: string): Promise<string> {
+    public async getValidSession(serviceProfile: IProfileLoaded,
```

Comment:
This code should extracted and made generic so that it works with other extenders as well. @JillieBeanSim had an implementation that was using the meta_data in the past.